tests/cmake_name_conflict/name_conflict.edl

```diff
@@ -4,7 +4,11 @@
 enclave {
     from "openenclave/edl/logging.edl" import oe_write_ocall;
     from "openenclave/edl/fcntl.edl" import *;
-    from "platform.edl" import *;
+#ifdef OE_SGX
```

Comment:
It seems like the user now has to add 
```c
#ifdef OE_SGX
    from "openenclave/edl/sgx/platform.edl" import *;
#else
    from "openenclave/edl/optee/platform.edl" import *;
#endif
```

Can we avoid the user from having to do this?

Reply:
After fully supporting system edl opt in, there will be more platform-dependent cases that can not simply rely on specifying path. For example,

#ifdef OE_SGX
    from "openenclave/edl/sgx/cpu.edl" import *;
     from "openenclave/edl/sgx/switchless.edl" import *;
#else
    from "openenclave/edl/optee/platform.edl" import *;
#endif

Given that, I think we need to use preprocessor (I thought this is the reason to support preprocessor, please correct me if I'm wrong).

Reply:
Note that this PR currently modifies tests only.

Reply:
With preprocessor, there are multiple options. For example here is an alternative.

openenclave/edl/platform.edl
```
#ifdef OE_SGX_FEATURE_CPUID
from "openenclave/edl/sgx/cpu.edl" import *;
#endif 

#ifdef OE_SGX_FEATURE_SWITCHLESS
from "openenclave/edl/sgx/switchless.edl" import *;
#endif 

#ifdef OE_TRUSTZONE
from "openenclave/edl/optee/platform.edl" import *;
#endif 
```

Then the user edl can just import openenclave/edl/platform.edl :
user.edl
```c
from "openenclave/edl/platform.edl" import *;
``` 
and select various features by specifying the appropriate defines:
```bash
oeedger8r  user.edl --search-path path-to-oe -DOE_SGX_FEATURE_SWITCHLESS -DOE_SGX_FEATURE_CPUID.
```

I'm not saying that this is the way to go. But it seems there are different user experiences possible.

Reply:
If we keep the import of `platform.edl` clean, we would cause the import of platform-dependent edls implicit. For example, in your example, it would be difficult for users to figure out the relationship between `-DOE_SGX_FEATURE_SWITCHLESS` and
```
#ifdef OE_SGX_FEATURE_SWITCHLESS
from "openenclave/edl/sgx/switchless.edl" import *;
#endif 
```
given that this import is hidden from users.

Reply:
Really my vision for this was that users would not import platform.edl. Instead they would just import the specific edl files they want for each platform they support.

I'm ok with leaving platform.edl as a convenience, but I agree with Ming-wei, I don't really want it to hide any more details than it already does.

Reply:
> If we keep the import of `platform.edl` clean, we would cause the import of platform-dependent edls implicit. For example, in your example, it would be difficult for users to figure out the relationship between `-DOE_SGX_FEATURE_SWITCHLESS` and
> 
> ```
> #ifdef OE_SGX_FEATURE_SWITCHLESS
> from "openenclave/edl/sgx/switchless.edl" import *;
> #endif 
> ```
> 
> given that this import is hidden from users.

In that case you'd have to document that flag `OE_SGX_FEATURE_SWITCHLESS` brings in switchless feature.
The intent of me bringing up that example was that there are multiple user experiences possible; not that I recommend one over the other.

So far I have not seen any documentation/discussion of the intended user experience. 
It is really not clear from this PR what the user experience is and I'd like our tests to mirror the user experience.
Hence I'm quite hesitant to approve this PR.

Reply:
> Really my vision for this was that users would not import platform.edl. Instead they would just import the specific edl files they want for each platform they support.
> 
> I'm ok with leaving platform.edl as a convenience, but I agree with Ming-wei, I don't really want it to hide any more details than it already does.

That's fair. But it is not clear how a user can figure out which edl files to import in the first place. It might be easy for us; but not quite obvious for the user.